example/service.go

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2020 AMIS Technologies
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"io/ioutil"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/getamis/alice/crypto/tss/dkg"
+	"github.com/getamis/alice/crypto/tss/message/types"
+	"github.com/getamis/sirius/log"
+	"github.com/gogo/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+// For simplicity, we use S256 curve in this example.
+var curve = btcec.S256()
+
+type service struct {
+	config *Config
+	pm     types.PeerManager
+
+	dkg  *dkg.DKG
+	done chan struct{}
+}
+
+func NewService(config *Config, pm types.PeerManager) (*service, error) {
+	s := &service{
+		config: config,
+		pm:     pm,
+		done:   make(chan struct{}),
+	}
+
+	// Create dkg
+	d, err := dkg.NewDKG(curve, pm, config.Threshold.DKG, config.Rank, s)
+	if err != nil {
+		log.Warn("Cannot create a new DKG", "config", config, "err", err)
+		return nil, err
+	}
+	s.dkg = d
+	return s, nil
+}
+
+func (p *service) Handle(s network.Stream) {
+	data := &dkg.Message{}
+	buf, err := ioutil.ReadAll(s)
+	if err != nil {
+		s.Reset()
```

Comment:
According the interface definition, 
```
// Reset closes both ends of the stream. Use this to tell the remote
// side to hang up and go away.
```
Do we want to disconnect if any error occur?